src/perception/filter.hpp

```diff
@@ -0,0 +1,91 @@
+#pragma once
+
+#include <algorithm>
+#include <array>
+#include <numeric>
+#include <vector>
+
+/***
+ * A filter that combines multiple readings into one.
+ * A user defined proportion acts as a median filter that gets rids of outliers,
+ * which are then piped into a mean filter that averages out the values.
+ *
+ * @tparam T Reading type
+ */
+template<typename T>
+class Filter {
```

Comment:
please give it a name, there is already more than one filter in our code

Reply:
Agreed